prometheus/counter.go

```diff
@@ -90,8 +91,12 @@ func NewCounter(opts CounterOpts) Counter {
 		nil,
 		opts.ConstLabels,
 	)
-	result := &counter{desc: desc, labelPairs: desc.constLabelPairs, now: time.Now}
+	if opts.now == nil {
+		opts.now = time.Now
+	}
+	result := &counter{desc: desc, labelPairs: desc.constLabelPairs, now: opts.now}
```

Comment:
Does "counter" type require "now"? We don't need to save it really, do we?

(Same with other metrics)

Reply:
Counters still need `now` for exemplars.

Reply:
Hw, can't find any usages of counter.now anywhere, what am I missing? 🤔

Reply:
Here it is: https://github.com/prometheus/client_golang/blob/51d24f86807c24ce866355d0c82258420f571c17/prometheus/counter.go#L170

😬